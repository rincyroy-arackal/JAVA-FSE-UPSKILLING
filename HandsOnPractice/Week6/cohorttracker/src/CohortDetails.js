import React from 'react';
import styles from './CohortDetails.module.css';

function CohortDetails({ cohort }) {
  const isOngoing = cohort.status.toLowerCase() === "ongoing";

  return (
    <div className={styles.box}>
      <h3 style={{ color: isOngoing ? "green" : "blue" }}>
        {cohort.name}
      </h3>
      <dl>
        <dt>Start Date:</dt>
        <dd>{cohort.startDate}</dd>

        <dt>End Date:</dt>
        <dd>{cohort.endDate}</dd>

        <dt>Status:</dt>
        <dd>{cohort.status}</dd>
      </dl>
    </div>
  );
}

export default CohortDetails;
