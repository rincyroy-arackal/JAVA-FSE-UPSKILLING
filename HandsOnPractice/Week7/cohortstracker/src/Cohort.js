export const CohortData = [
  {
    code: 'CT2024',
    name: 'React Bootcamp',
    status: 'Ongoing'
  },
  {
    code: 'CT2023',
    name: 'Angular Sprint',
    status: 'Completed'
  }
];
