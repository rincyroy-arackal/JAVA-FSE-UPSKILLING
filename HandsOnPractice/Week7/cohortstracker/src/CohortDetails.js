import React from 'react';

function CohortDetails(props) {
  const cohort = props.cohort || { code: 'N/A', name: 'N/A', status: 'N/A' };

  return (
    <div>
      <h3>{cohort.code}</h3>
      <p>Name: {cohort.name}</p>
      <p>Status: {cohort.status}</p>
    </div>
  );
}

export default CohortDetails;
