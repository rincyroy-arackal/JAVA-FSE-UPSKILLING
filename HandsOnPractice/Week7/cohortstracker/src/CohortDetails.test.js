import React from 'react';
import { render, screen } from '@testing-library/react';
import CohortDetails from './CohortDetails';
import { CohortData } from './Cohort';

describe('CohortDetails component', () => {
  test('renders without crashing', () => {
    render(<CohortDetails />);
    expect(screen.getByRole('heading')).toHaveTextContent('N/A');
    expect(screen.getByText('Name: N/A')).toBeInTheDocument();
    expect(screen.getByText('Status: N/A')).toBeInTheDocument();
  });

  test('displays cohort data passed via props', () => {
    const cohort = CohortData[0];
    render(<CohortDetails cohort={cohort} />);
    expect(screen.getByRole('heading')).toHaveTextContent(cohort.code);
    expect(screen.getByText(`Name: ${cohort.name}`)).toBeInTheDocument();
    expect(screen.getByText(`Status: ${cohort.status}`)).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const cohort = CohortData[0];
    const { asFragment } = render(<CohortDetails cohort={cohort} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
