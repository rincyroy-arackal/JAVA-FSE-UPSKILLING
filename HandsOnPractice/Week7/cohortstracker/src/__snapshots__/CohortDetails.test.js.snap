// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CohortDetails component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      CT2024
    </h3>
    <p>
      Name: React Bootcamp
    </p>
    <p>
      Status: Ongoing
    </p>
  </div>
</DocumentFragment>
`;
